src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerRegistryEndpoint.java

```diff
@@ -236,6 +236,9 @@ public String imageName(String userAndRepo) throws IOException {
         if (effectiveUrl.getPort() > 0 && effectiveUrl.getDefaultPort() != effectiveUrl.getPort()) {
             s.append(':').append(effectiveUrl.getPort());
         }
+        if (userAndRepo.startsWith(String.valueOf(s))) {
```

Comment:
Before doing it you need to ensure the field is not null.
There is no related annotations in the code. Maybe makes sense to update method definition